pep-0581.rst

```diff
@@ -0,0 +1,310 @@
+PEP: 581
+Title: Using GitHub Issues for CPython
+Version: $Revision$
+Last-Modified: $Date$
+Author: Mariatta Wijaya <mariatta@python.org>
+Status: Draft
+Type: Process
+Content-Type: text/x-rst
+Created: 20-Jun-2018
+
+
+Abstract
+========
+
+This PEP outlines the steps required to migrate Python's issue tracker
+from Roundup to using GitHub Issues.
+
+
+Rationale
+=========
+
+CPython's development moved to GitHub on February 2017. All other projects within
+The PSF's organization are hosted on GitHub and are using GitHub issues.
+CPython is still using Roundup as the issue tracker in bugs.python.org (bpo).
+
+Why GitHub
+----------
+
+It has a lot of nice features readily available out of the box, that are not
+currently available in Roundup / bpo.
+
+- APIs we can use to build integrations and automations.
+
+- Ability to attach screenshots and files in pull requests and issues.
+
+- Administrators and core developers can edit issues comments and pull requests.
+
+- Ability to reply to issues and pull request conversations via email.
+
+- It supports two factor authentication.
+
+- It supports markdown and emoji.
+
+- Core developers don't have to maintain it, giving us more time to focus on
+  the development of Python.
+
+- We can automatically close issues when a PR has been merged [#close-github-issue]_.
+
+- With more than 28 million users, an open source contributor is more likely
+  to already have an account, and familiar with GitHub interface, making it
+  easier to start contributing.
+
+- Email notifications contain metadata [#email-meta]_, integrated with GMail, and
+  allows you to systematically filter emails.
+
+- Email address can remain hidden by choice.
+
+
+Issues with bpo / Roundup
+-------------------------
+
+- It is not actively maintained.
+
+- Less than five people maintain bpo. Some of them are core developers.
+
+- It is in mercurial. Without any CI available, it puts heavy burden on existing
+  (few) maintainers in terms of reviewing, testing, and applying patches.
+
+- At its current state, it is not equipped to accept lots of contributions from
+  people who aren't already familiar with the code base.
+
+- The upstream roundup is in mercurial. If bpo does move to GitHub, it will become
+  difficult to update patches from upstream.
+
+- The user interface needs update and redesign. It will require UX/UI research
+  to keep it up to date with current web standards.
+
+- Email address is exposed. There is no choice to mask it.
+
+- There is no REST API available. There is an open issue in Roundup for adding
+  REST API  [#roundup-rest-api]_. Last activity was in 2016.
+
+- I personally don't want to receive email each time someone added themselves as "nosy"
```

Comment:
You might express it without writing "I", maybe: Email notifications are sent each time someone adds themself as "nosy" and it's not possible to opt-out for such notification.

In general, Roundup sends many useless notifications and it's not possible or hard to configure notifications. GitHub has a global "Ignore" option per project, and a similar thing to "nosy": Subscribe/Unbscribe per issue. But no notification to everybody when someone subscribes (good!).